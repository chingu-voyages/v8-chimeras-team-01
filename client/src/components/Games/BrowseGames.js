import React, { useState, useEffect } from 'react';

export default function BrowseGames({ showGames, handleToggleGames }) {

  /**
   * @method [useState(gameList)]
   * @description [Array of games available to map over and display in tiles]
   */
  const [gameList, setgameList] = useState([
    {
      id: "",
      name: "",
      image: <i className="fas fa-stroopwafel fa-6x"></i>,
      desc: "What you know 'bout Waffles??",
    }
  ]);

  useEffect(() => {
    async function fetchQuestions() {
    const response = await fetch('/api/questions');
    const data = await response.json();
    return setgameList({
      id: data[0].id,
      name: data[0].category,
      image: <i className="fas fa-stroopwafel fa-6x"></i>,
      desc: "What you know 'bout Waffles??"
    })
  }
    fetchQuestions();
  }, []);

  return (
    <section className={!!showGames ? 'games show-games' : "games"} >
      <button className="games-close pm0 pointy"
              onClick={() => {handleToggleGames(false)}}> X close </button>
            <h1>Browse Available Challenges.</h1>
      <div className="games-container">
          <article className="game-tile" key={gameList.id} >
            <div className="gt-image fbc pm0">{gameList.image}</div>
            <div className="gt-description pm0">
              <h4>{gameList.name}</h4>
              <p>{gameList.desc}</p>
            </div>
          </article>
      </div>
    </section>
  )
}
