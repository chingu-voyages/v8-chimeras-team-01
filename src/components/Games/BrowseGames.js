import React, { useState } from 'react';

export default function BrowseGames({ showGames, handleToggleGames }) {

  /**
   * @method [useState(gameList)]
   * @description [Array of games available to map over and display in tiles]
   */
  const [gameList] = useState([
    {
      id: "0001",
      name: "Waffles",
      image: <i className="fas fa-stroopwafel fa-6x"></i>,
      desc: "What you know 'bout Waffles??",
    }
  ]);

  return (
    <section className={!!showGames ? 'games show-games' : "games"} >
      <button className="games-close pm0 pointy"
              onClick={() => {handleToggleGames(false)}}> X close </button>
            <h1>Browse Available Challenges.</h1>
      <div className="games-container">
        {gameList.map(game => (
          <article className="game-tile" key={game.id} >
            <div className="gt-image fbc pm0">{game.image}</div>
            <div className="gt-description pm0">
              <h4>{game.name}</h4>
              <p>{game.desc}</p>
            </div>
          </article>
        ))}
      </div>
    </section>
  )
}
