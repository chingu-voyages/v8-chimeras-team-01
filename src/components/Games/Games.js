import React from 'react';
import './index.css';

export default function Games({gameList}) {

  return (
    <section id='games'>
      <h1>Choose your challenge</h1>
      {gameList.map(game => (
        <article className="game-tile" key={game.id}>
          <div className="gt-image">{game.image}</div>
          <div className="gt-description">
            <h4>{game.name}</h4>
            <p>{game.desc}</p>
          </div>
        </article>
      ))}
    </section>
  )
}


// {
//   id: "0001",
//   name: "Waffles",
//   image: `<i class="fas fa-stroopwafel"></i>`,
//   desc: "What you know 'bout Waffles??"
// }
