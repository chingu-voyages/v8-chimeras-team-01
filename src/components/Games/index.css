@import '../../App.css';

.games {
  position: absolute;
  top: 1000px;
  opacity: 0;
  width: 100%;
  background-color: #fff;
  transition-duration: 400ms;
}

.show-games {
  top: 100px;
  opacity: 1
}

.games>h1 {
  width: 100%;
  margin-left: 0.3em;
  font-family: var(--font-header);
  font-size: 2em;
}

.games-close {
  position: absolute;
  top: 10px;
  right: 0;
  background-color: #fff;
  border: none;
  width: 100px;
  padding: 0;
  margin: 0;
}

.games-container {
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(200px, 1fr));
  grid-gap: 15px;
}

.game-tile:hover .gt-image {
  background-color: #000;
  color: #fff;
}

.game-tile {
  width: 200px;
  height: 250px;
  margin: 1em auto;
  box-shadow: 2px 1px 8px var(--color-gray);
}

.gt-image {
  width: 200px;
  height: 130px;
  padding: 0;
  margin: 0;
  background-color: var(--color-yellow);
  color: #000;
}

.gt-description {
  box-sizing: border-box;
  width: 200px;
  height: 120px;
  padding: 0;
  margin: 0;
  text-align: center;
}

.gt-description>* {
  padding: 0.5em;
  margin: 0;
}
