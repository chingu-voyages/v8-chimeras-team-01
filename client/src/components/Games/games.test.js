import React from 'react';
import { shallow } from 'enzyme';

import Games from './Games.js';

describe('Games', () => {

  it('should render correctly', () => {
    const component = shallow(<Games />);
    expect(component).toMatchSnapshot();
  });

});
