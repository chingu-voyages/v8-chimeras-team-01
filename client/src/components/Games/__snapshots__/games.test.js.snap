// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Games should render correctly 1`] = `
<section
  className="games"
>
  <button
    className="games-close pm0 pointy"
    onClick={[Function]}
  >
     X close 
  </button>
  <h1>
    Choose your challenge
  </h1>
  <div
    className="games-container"
  >
    <article
      className="game-tile"
      key="0001"
    >
      <div
        className="gt-image fbc pm0"
      >
        <i
          className="fas fa-stroopwafel fa-6x"
        />
      </div>
      <div
        className="gt-description pm0"
      >
        <h4>
          Waffles
        </h4>
        <p>
          What you know 'bout Waffles??
        </p>
      </div>
    </article>
    <article
      className="game-tile"
      key="0002"
    >
      <div
        className="gt-image fbc pm0"
      >
        <i
          className="fas fa-cookie fa-6x"
        />
      </div>
      <div
        className="gt-description pm0"
      >
        <h4>
          Cookies
        </h4>
        <p>
          You know you can eat them, but what else??
        </p>
      </div>
    </article>
    <article
      className="game-tile"
      key="0003"
    >
      <div
        className="gt-image fbc pm0"
      >
        <i
          className="fas fa-stroopwafel fa-6x"
        />
      </div>
      <div
        className="gt-description pm0"
      >
        <h4>
          Waffles
        </h4>
        <p>
          What you know 'bout Waffles??
        </p>
      </div>
    </article>
    <article
      className="game-tile"
      key="0004"
    >
      <div
        className="gt-image fbc pm0"
      >
        <i
          className="fas fa-cookie fa-6x"
        />
      </div>
      <div
        className="gt-description pm0"
      >
        <h4>
          Cookies
        </h4>
        <p>
          You know you can eat them, but what else??
        </p>
      </div>
    </article>
  </div>
</section>
`;
